src/System.Management.Automation/engine/serialization.cs

```diff
@@ -4039,38 +4053,78 @@ private object ReadDictionary(ContainerType ct)
 
                     object value = ReadOneObject();
 
-                    // On the first collision, copy the hash table to one that uses the default comparer.
-                    if (table.ContainsKey(key) && (keyClashFoundIteration == 0))
+                    if (!isOrdered)
                     {
-                        keyClashFoundIteration++;
-                        Hashtable newHashTable = new Hashtable();
-                        foreach (DictionaryEntry entry in table)
+                        // On the first collision, copy the hash table to one that uses the default comparer.
+                        if (table.ContainsKey(key) && (keyClashFoundIteration == 0))
```

Comment:
```suggestion
                        if (keyClashFoundIteration == 0 && table.ContainsKey(key))
```

Below too.

Reply:
This was a copy from the original code - changed in all 4 places (not/ordered x iteration=0/1).